plugins/plugin_utils/ansible_type.py

```diff
@@ -42,21 +44,21 @@ def _ansible_type(data, alias, *, use_native_type: bool = False):
         alias = {}
 
     if not isinstance(alias, Mapping):
-        msg = "The argument alias must be a dictionary. %s is %s"
-        raise AnsibleFilterError(msg % (alias, type(alias)))
+        msg = f"The argument alias must be a dictionary. {alias!r} is {type(alias)}"
+        raise AnsibleFilterError(msg)
 
     data_type = _atype(data, alias, use_native_type=use_native_type)
 
     if data_type == "list" and len(data) > 0:
-        items = [_atype(i, alias, use_native_type=use_native_type) for i in data]
-        items_type = "|".join(sorted(set(items)))
+        items = {_atype(i, alias, use_native_type=use_native_type) for i in data}
+        items_type = "|".join(sorted(items))
         return f"{data_type}[{items_type}]"
 
     if data_type == "dict" and len(data) > 0:
-        keys = [_atype(i, alias, use_native_type=use_native_type) for i in data.keys()]
-        vals = [_atype(i, alias, use_native_type=use_native_type) for i in data.values()]
-        keys_type = "|".join(sorted(set(keys)))
-        vals_type = "|".join(sorted(set(vals)))
+        keys = {_atype(i, alias, use_native_type=use_native_type) for i in data.keys()}
+        vals = {_atype(i, alias, use_native_type=use_native_type) for i in data.values()}
+        keys_type = "|".join(sorted(keys))
+        vals_type = "|".join(sorted(vals))
         return f"{data_type}[{keys_type}, {vals_type}]"
```

Comment:
Not sure what the semantics are in this part but, whilst `keys` will always be unique, that is not guaranteed for `vals`, so either:
* `set()` has to be maintained, potentially resulting in "|-delimited-lists-as-string" of different lengths in `keys_type` and `vals_type` - liekly to be wrong
* that risk always existed before, so this is a bug fix - if that's the case, then it would take some generosity to fit this under "fix parameter checking" - it is more the handling of the parameter _after_ it was (allegedly) validated than the validation itself. Nit picking, I know.

Reply:
That's intentional and how the plugin has been working since it was created in this repo.

Reply:
Glancing at the code - not diving into this one - it looks very odd and potentially a bug. But yes, it's how it has been working since forever.